generator/osm2meta.cpp

```diff
@@ -126,6 +128,169 @@ string MetadataTagProcessorImpl::ValidateAndFormat_url(string const & v) const
   return v;
 }
 
+string MetadataTagProcessorImpl::ValidateAndFormat_facebook(string const & facebookPage) const
+{
+  if (facebookPage.empty())
+    return {};
+  //check that facebookPage contains valid username. See rules: https://www.facebook.com/help/105399436216001
+  if (strings::EndsWith(facebookPage, ".com") || strings::EndsWith(facebookPage, ".net"))
+    return {};
+  if (regex_match(facebookPage, regex(R"(^@?[a-zA-Z\d.\-]{5,}$)")))
+  {
+    if (facebookPage.front() == '@')
+      return facebookPage.substr(1);
+    else
+      return facebookPage;
+  }
+  if (EditableMapObject::ValidateWebsite(facebookPage))
+  {
+    string facebookPageUrl = facebookPage;
+    //Check if HTTP protocol is present
+    if (!strings::StartsWith(facebookPage, "http://") && !strings::StartsWith(facebookPage, "https://"))
+      facebookPageUrl = "https://" + facebookPage;
+
+    const url::Url url = url::Url(facebookPageUrl);
+    const string &domain = strings::MakeLowerCase(url.GetWebDomain());
+    // Check Facebook domain name
+    if (strings::EndsWith(domain, "facebook.com") || strings::EndsWith(domain, "fb.com")
+       || strings::EndsWith(domain, "fb.me") || strings::EndsWith(domain, "facebook.de")
+       || strings::EndsWith(domain, "facebook.fr"))
+    {
+      auto webPath = url.GetWebPath();
+      //Strip last '/' symbol
+      if (webPath.back() == '/')
+        return webPath.substr(0, webPath.length()-1);
+      else
+        return webPath;
+    }
+  }
+
+  return {};
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_instagram(string const & instagramPage) const
+{
+  if (instagramPage.empty())
+    return {};
+  //check that instagramPage contains valid username. Rules took here: https://blog.jstassen.com/2016/03/code-regex-for-instagram-username-and-hashtags/
```

Comment:
Nit1: // Check…
Nit2: Start the second sentence from a new line to better fit the screen.

Reply:
Fixed formatting